system/Database/MySQLi/Result.php

```diff
@@ -95,11 +131,12 @@ public function getFieldData(): array
 			$retVal[$i]              = new \stdClass();
 			$retVal[$i]->name        = $data->name;
 			$retVal[$i]->type        = $data->type;
+			$retVal[$i]->typeName    = isset($data_types[$data->type]) ? $data_types[$data->type] : $data->type;
```

Comment:
This should be `null` if key is missing.

Reply:
Will change to be null then, but to be honest I believe the developer might more quickly find the reason for it returning a number instead of the text if he gets a number as n reply and realize it is not mapped. But will make the change as requested.